api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -55,11 +56,24 @@ module.exports = {
     const trx = await Bookshelf.knex.transaction();
 
     try {
+      const nationalStudentIdsFromFile = schoolingRegistrationDatas.map((schoolingRegistrationData) => schoolingRegistrationData.nationalStudentId);
+      const students = await studentRepository.findReconciledStudentsByNationalStudentId(nationalStudentIdsFromFile);
 
       const schoolingRegistrations = _.map(schoolingRegistrationDatas, (schoolingRegistrationData) => new SchoolingRegistration({ ...schoolingRegistrationData, organizationId }));
       const schoolingRegistrationsFromOrganization = await this.findByOrganizationId({ organizationId });
-      const nationalStudentIdsFromOrganization = _.map(schoolingRegistrationsFromOrganization, 'nationalStudentId');
-      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => _.includes(nationalStudentIdsFromOrganization, schoolingRegistration.nationalStudentId));
+
+      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => {
+        const schoolingRegistrationFromOrganization = schoolingRegistrationsFromOrganization.find((schoolingRegistrationFromOrganization) => schoolingRegistrationFromOrganization.nationalStudentId === schoolingRegistration.nationalStudentId);
```

Comment:
Quand je lis `const schoolingRegistrationFromOrganization = schoolingRegistrationsFromOrganization.` l'alarme a nommage de variable se déclenche.. Je ne saisis pas tout, mais que pensez-vous de 

```suggestion
        const reconciledSchoolingRegistration = schoolingRegistrationsFromOrganization.find((schoolingRegistrationFromOrganization) => schoolingRegistrationFromOrganization.nationalStudentId === schoolingRegistration.nationalStudentId);
```

Reply:
C'est après qu'on vérifie s'il est reconcilié. Ici on cherche juste sur la personne est déjà enregistré dans l'organisation.